packages/web-react/src/components/Grid/useGridStyleProps.ts

```diff
@@ -34,44 +17,18 @@ export interface GridStyles<T> {
 }
 
 export function useGridStyleProps(props: SpiritGridProps<ElementType>): GridStyles<SpiritGridProps<ElementType>> {
-  const { cols, ...restProps } = props;
+  const { cols, spacing, spacingX, spacingY, ...restProps } = props;
 
   const gridClass = useClassNamePrefix('Grid');
 
-  const gridStyle: GridCSSProperties = {};
-
-  const typePropNames = Object.keys(props).filter((propName) => propName.startsWith('spacing'));
-
-  typePropNames.forEach((propName) => {
-    let type: string;
-
-    if (props[propName]) {
-      if (propName.startsWith('spacingX')) {
-        type = 'spacing-x';
-      } else if (propName.startsWith('spacingY')) {
-        type = 'spacing-y';
-      } else {
-        type = propName;
-      }
-
-      if (type === 'spacing') {
-        setGridStyle(
-          gridStyle,
-          `grid-spacing-x${propName.replace(type, '').toLowerCase()}`,
-          props[propName] as GridCSSProperties,
-        );
-        setGridStyle(
-          gridStyle,
-          `grid-spacing-y${propName.replace(type, '').toLowerCase()}`,
-          props[propName] as GridCSSProperties,
-        );
-      } else {
-        setGridStyle(gridStyle, `grid-${type}`, props[propName] as GridCSSProperties);
-      }
-    }
-
-    delete props[propName];
-  });
+  const gridStyle: GridCSSProperties = {
+    ...{
+      ...useSpacingStyle(spacing, 'grid', 'x'),
```

Comment:
```suggestion
      ...useSpacingStyle(spacing, 'grid', Spacing.VERTICAL),
```

Can we define some constants for this, please? This looks a little bit magically :-)